### PR TITLE
HPCC-21404 Prompt users for confirmation before proceeding with purgi…

### PR DESCRIPTION
**Pull request by gfortil**
…ng the platform

Signed-off-by: Fortil, Godson <godson.fortil@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ x] This change is a new feature (non-breaking change which adds functionality).
- [ x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ x] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x ] My code follows the code style of this project.
  - [ x] My code does not create any new warnings from compiler, build system, or lint.
- [ x] The commit message is properly formatted and free of typos.
  - [ x] The commit message title makes sense in a changelog, by itself.
  - [x ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ x] Any new interfaces or exported functions are appropriately commented.
- [ x] I have read the CONTRIBUTORS document.
- [ x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [x ] Performance
  - [x ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ x] The target branch of this pull request is appropriate for such a change.
- [x ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ x] Send notifications about my Pull Request position in Smoketest queue.
- [ x] Test my draft Pull Request.

### CODE REVIEW

**Comment by gfortil** [comment]
@Michael-Gardner could you please review this?

**Comment by Michael-Gardner** [comment]
Reviewing now

**Comment by richardkchapman** [comment]
@Michael-Gardner Are you still reviewing?

File: initfiles/sbin/complete-uninstall.sh.in [hunk 1]
```diff
@@ -83,87 +85,98 @@ while true ; do
...
 
-set_environmentvars
+pckname=$(apt-cache pkgnames hpccsystems-platform 2> /dev/null)
 
-mklink=""
-lpath=$(canonicalize_path ${path})
-if [ -n "${lpath}" -a -h "${lpath}" ] ; then
-    mklink=$(readlink "${lpath}")
-    # echo "\"${lpath}\" is a soft-link to \"${mklink}\""
+if [ $purge != 'purge' ];then
```
Comment (Michael-Gardner) [inline initfiles/sbin/complete-uninstall.sh.in:99]: This line needs to be [ "$purge" != "purge" ]  or else you'll get a unary operator error for the != if purge isn't populated.

File: initfiles/sbin/complete-uninstall.sh.in [hunk 2]
```diff
@@ -83,87 +85,98 @@ while true ; do
...
 
-if [ -e /etc/debian_version ]; then
-    pckname=$(apt-cache pkgnames hpccsystems-platform 2> /dev/null)
-    if [ -n "${pckname}" ] ; then
-        echo "Removing ${pckname}"
-        apt-get remove -y ${pckname}
-        if [ $? -ne 0 ]; then
-           echo "An error has occured during removal.  Try \`apt-get -f remove ${pckname}\` to fix this issue."
-           exit 1
+if [ $purge = "purge" ]; then
```
Comment (Michael-Gardner) [inline initfiles/sbin/complete-uninstall.sh.in:103]: Same issue here.

 [ "$purge" != "purge" ]

**Comment by gfortil** [comment]
@Michael-Gardner please review again

File: initfiles/sbin/complete-uninstall.sh.in [hunk 3]
```diff
@@ -83,87 +85,98 @@ while true ; do
...
+    fi
+
+    if [ -e /etc/debian_version ]; then
+        if [ -n "${pckname}" ] ; then
+            apt-get remove -y ${pckname}
+            if [ $? -ne 0 ]; then
+            echo "An error has occured during removal.  Try \`apt-get -f remove ${pckname}\` to fix this issue."
+            exit 1
+            fi
+            if [ $1 -eq 0 ]; then
```
Comment (Michael-Gardner) [inline initfiles/sbin/complete-uninstall.sh.in:120]: this line should be

`if [ $leaveenv -eq 0 ]; then`

**Review by gfortil — COMMENTED** [review]
done @Michael-Gardner

**Comment by Michael-Gardner** [comment]
@richardkchapman ready for merge

**Pull request merged**